### PR TITLE
Adding ability to have as many command outputs go to meta data as needed.

### PR DESCRIPTION
**Pull request by zhilbug**
Adding an overridable git describe command setting the standard to just the tag name.

Adding option to add this as the Assembly Version which is off by default.

Updating readme to reflect changes.

### CODE REVIEW

File: MSBuildGitHash/build/MSBuildGitHash.targets [hunk 1]
```diff
@@ -2,7 +2,9 @@
 
   <PropertyGroup>
     <IncludeMSBuildGitHashMetadata Condition="'$(IncludeMSBuildGitHashMetadata)' == ''">true</IncludeMSBuildGitHashMetadata>
+    <IncludeMSBuildGitDescribeMetadata Condition="'$(IncludeMSBuildGitDescribeMetadata)' == ''">true</IncludeMSBuildGitDescribeMetadata>
     <IncludeMSBuildGitHashInfoVersion Condition="'$(IncludeMSBuildGitHashInfoVersion)' == ''">true</IncludeMSBuildGitHashInfoVersion>
+    <IncludeMSBuildGitDescribeAssemblyVersion Condition="'$(IncludeMSBuildGitDescribeAssemblyVersion)' == ''">false</IncludeMSBuildGitDescribeAssemblyVersion>
```
Comment (drdamour) [inline MSBuildGitHash/build/MSBuildGitHash.targets]: this is kinda weird cause it's unclear what would happen if this and IncludeMSBuildGitHashInfoVersion  were both true, i wasn't expecting to go this far

File: MSBuildGitHash/build/MSBuildGitHash.targets [hunk 2]
```diff
@@ -103,22 +129,30 @@
         Include="System.Reflection.AssemblyInformationalVersionAttribute">
         <_Parameter1>$(MSBuildGitHashVersionAttribute)</_Parameter1>
       </AssemblyAttributes>
+      <AssemblyAttributes
```
Comment (drdamour) [inline MSBuildGitHash/build/MSBuildGitHash.targets]: same as above, i think this is too far

File: MSBuildGitHash/build/MSBuildGitHash.targets (hunk 1 above)
Comment (zhilbug) [reply MSBuildGitHash/build/MSBuildGitHash.targets]: So add it to the custom attributes but NOT the assembly version meta data?

Comment (drdamour) [reply MSBuildGitHash/build/MSBuildGitHash.targets]: yeah cause u can only have 1 be the assembly version...i think a further enhancement would be to deprecate IncludeMSBuildGitHashInfoVersion  and instead have it be more of a map as to what value should be used maybe

Comment (drdamour) [reply MSBuildGitHash/build/MSBuildGitHash.targets]: but dont' want to do anything related to that now

**Comment by MarkPflug** [comment]
I've only looked very briefly at this PR.

I thought a bit about #20, and my initial thought was to make the mechanism more general purpose. Rather than hard-code every special case (git hash, git version, etc) using msbuild properties, use an itemgroup instead:

```
<ItemGroup>
<AssemblyEmbed Include="GitHash">
    <Command>git describe --long --always --dirty --exclude=* --abbrev=7</Command>
</AssemblyEmbed>
<AssemblyEmbed Include="GitDescription">
    <Command>git describe --tag</Command>
</AssemblyEmbed>
</ItemGroup>
```

The GetGitHash target would process these using [batching](https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild-batching?view=vs-2019) and include the command output as metadata. Then, the GenerateAssemblyAttributes would process that batch to create the AssemblyAttributes item group. This should be possible to do this and minimize the potential to break existing builds, (ie `<Command>$(MSBuildGitHashCommand)</Command>`).

I'm not going to have time soon to deal with this, so if this is urgent I would recommend forking and creating your own package, or putting this into a [Directory.Build.targets](https://docs.microsoft.com/en-us/visualstudio/msbuild/customize-your-build?view=vs-2019) for your solution.

**Comment by drdamour** [comment]
@MarkPflug how would a user of the package modify the command?

**Comment by MarkPflug** [comment]
The same way they do now, for the existing commands: override the property.
For items in the new model:
```
<AssemblyEmbed Update="GitDescription">
<Command>[custom command here]</Command>
```

**Comment by MarkPflug** [comment]
Now that I see the beginnings of an implementation, I see the problem:
Using the `Exec` task with a batched item group (AssemblyEmbed), I don't see a way to associate the output (exitcode, consoleoutput) back to the item that invoked it.

This might require a custom task, which would require an assembly be included. I'm okay with an assembly being included as a build time dependency, just not one that would be a runtime dependency.

My thought was that an itemGroup would be defined (Which could be customized by consumers):
```
<ItemGroup>
<AssemblyEmbed Include="GitHash">
    <Command>git describe --long --always --dirty --exclude=* --abbrev=7</Command>
    <ValidationRegex>SOMEREGEX</ValidationRegex>
</AssemblyEmbed>
<AssemblyEmbed Include="GitDescription">
    <Command>git describe --tag</Command>
</AssemblyEmbed>
```

The new task (which would replace what Exec does now) would process each item in this group, invoke the command, capture the output and exit code, validate the output, and then embellish the original itemgroup with metadata, so it would look like:

```
<ItemGroup>
<AssemblyEmbed Include="GitHash">
    <Command>git describe --long --always --dirty --exclude=* --abbrev=7</Command>
    <ValidationRegex>SOMEREGEX</ValidationRegex>
    <ExitCode>0</ExitCode>
    <ConsoleOutput>ABCDEF</ConsoleOutput>
</AssemblyEmbed>
<AssemblyEmbed Include="GitDescription">
    <Command>git describe --tag</Command>
    <ExitCode>0</ExitCode>
    <ConsoleOutput>MyFavoriteRelease</ConsoleOutput>

</AssemblyEmbed>
```
Then, a subsequent task would process this into the assembly metadata. The issue is that build-in `Exec` task doesn't allow this to be done, or at least I can't see a way.

Do think this is worth investigating? I'm not sure the extra complexity is worth it.

**Comment by MarkPflug** [comment]
Okay, maybe it is possible. MSBuild (while crazy sometimes) always amazes me at what is possible:

Here is a POC: 
test.proj
```
<Project DefaultTarget="Build">

    <ItemGroup>
        <AssemblyEmbed Include="GitHash">
            <Command>cd</Command>
        </AssemblyEmbed>
        <AssemblyEmbed Include="GitDescription">
            <Command>echo Example text</Command>
        </AssemblyEmbed>
        <AssemblyEmbed Include="Somethingelse">
            <Command>FAILFAILFAIL</Command>
        </AssemblyEmbed>
    </ItemGroup>

    <Target Name="Build" DependsOnTargets="CmdBatch">

        <Warning Text="%(AssemblyEmbed.Identity) %(AssemblyEmbed.Command) (%(AssemblyEmbed.ExitCode)) %(AssemblyEmbed.ConsoleOutput)"/>

    </Target>

    <Target Name="CmdBatch" Inputs="@(AssemblyEmbed)" Outputs="%(Identity)">
        <Exec 
            Command="%(AssemblyEmbed.Command)"
            ConsoleToMsBuild="true"
            IgnoreExitCode="true"
            >
            <Output PropertyName="ConsoleOutput" TaskParameter="ConsoleOutput"/>
            <Output PropertyName="ExitCode" TaskParameter="ExitCode"/>
        </Exec>

        <ItemGroup>
            <AssemblyEmbed Update="%(Identity)">
                <ConsoleOutput>$(ConsoleOutput)</ConsoleOutput>
                <ExitCode>$(ExitCode)</ExitCode>
            </AssemblyEmbed>
        </ItemGroup>
    </Target>

</Project>
```

Does this make sense @zhilbug? Do you want to run with it?

Edit: sorry, it probably wasn't obvious that the ` Inputs="@(AssemblyEmbed)" Outputs="%(Identity)"` is the key to making the CmdBatch Target run per-item against the `AssemblyEmbed` group. This way `Update="%(Identity)"` can be used to push metadata back into the original item.

**Comment by zhilbug** [comment]
@MarkPflug this does make sense and solves the issue that I was having! I will run with it and post some updates today. I really like your idea with this!

**Comment by zhilbug** [comment]
@MarkPflug Let me know if I missed something you were thinking here or would like me to change.

**Comment by MarkPflug** [comment]
Thank you @zhilbug!
I will try to find some time in the next few days to thoroughly review this. At first glance it looks like what I was expecting.

**Comment by MarkPflug** [comment]
I merged this. It required a bit of adjustment to keep all the unit tests passing and a few modifications to maintain backwards compatibility.
I pushed 2.0.0-B0001 (prerelease) package to nuget. If you can test it and make sure it suits your needs, I'll then push a full 2.0.0. release.

Thanks again for your work on this.

**Comment by zhilbug** [comment]
@MarkPflug Thank you so much for being willing to work with us on this.  We greatly appreciate it!  I have tested it for our needs and it works flawlessly.

**Comment by MarkPflug** [comment]
No problem. There is one change I need to roll back before I release 2.0. I also thought it was a good idea to add the build time as an assembly metadata by default (you might see this in your assemblies). I realized this was a bad idea, because it will break deterministic builds. I'll try to get a 2.0 release out this afternoon.

**Comment by zhilbug** [comment]
Got it, sounds good.  Just let me know.  You rock!

**Comment by MarkPflug** [comment]
2.0 release package should be listed now.

**Pull request merged**